app/Main.hs

```diff
@@ -38,19 +53,53 @@ printChecks checks = do
 
 parseOptions :: Parser LintOptions
 parseOptions =
-    LintOptions <$> switch (long "version" <> short 'v' <> help "Show version") <*>
-    many (strOption (long "ignore" <> help "Ignore rule" <> metavar "RULECODE")) <*>
+    LintOptions <$> -- CLI options parser definition
+    switch (long "version" <> short 'v' <> help "Show version") <*>
+    many
+        (strOption
+             (long "ignore" <> help "Ignore rule. If present, config file is ignored" <>
+              metavar "RULECODE")) <*>
     many (argument str (metavar "DOCKERFILE..."))
 
 main :: IO ()
-main = execParser opts >>= lint
+main = execParser opts >>= applyConfig >>= lint
   where
     opts =
         info
             (helper <*> parseOptions)
             (fullDesc <> progDesc "Lint Dockerfile for errors and best practices" <>
              header "hadolint - Dockerfile Linter written in Haskell")
 
+applyConfig :: LintOptions -> IO LintOptions
+applyConfig o@(LintOptions _ (_:_) _) = return o
+applyConfig o = do
+    theConfig <- findConfig
+    case theConfig of
+        Nothing -> return o
+        Just config -> parseAndApply config
+  where
+    findConfig = do
+        localConfigFile <- (</> "hadolint.yaml") <$> getCurrentDirectory
+        configFile <- getXdgDirectory XdgConfig "hadolint.yaml"
```

Comment:
Would you mind to have it hidden by defaul, eg `.hadolint.yaml`?

Reply:
Support both hidden and not hidden?

Reply:
I think hidden(starting with `.`) like `.travis.yml` or `.hindent.yaml` or `.zshrc` is enough.